### ISSUE TITLE
Version v4.7.1 Lambda Layer Misconfigured: All Regions Pointing to us-east-1

### ISSUE THREAD

**Issue by changboyu** [elastic/apm-agent-nodejs#4171]
**Describe the bug**

I found that the ARN table for Lambda Layer version v4.7.1 is incorrect. All regions are pointing to the us-east-1 region, as depicted in the attached image. 

[Release page](https://github.com/elastic/apm-agent-nodejs/releases/tag/v4.7.1)

<img width="1666" alt="image" src="https://github.com/user-attachments/assets/f4bc2702-7f49-42ed-a728-6a5fa834ef75">

This impact page [Monitoring AWS Lambda Node.js Functions](https://www.elastic.co/guide/en/apm/agent/nodejs/current/lambda.html) 

<img width="938" alt="image" src="https://github.com/user-attachments/assets/2d01d404-23de-44ce-90d0-4e1a84ee9aaa">

**To Reproduce**

1. Visit [Monitoring AWS Lambda Node.js Functions](https://www.elastic.co/guide/en/apm/agent/nodejs/current/lambda.html) stop to load the lambda layer arn for other regions.
2. Choose a region rather than us-east-1
3. The guide's APM agent layer is missing and shows <SELECTED_AGENT_LAYER_ARN>

**Expected behavior**
Each Region should have their own lambda layer
</details>

**Comment by trentm** [comment]
Thanks for reporting this! I think I've found the issue. We'll try to get a new release out in the next few days.

**Comment by trentm** [comment]
The v4.7.1 release published the lambda layers incorrectly. Instead of
publishing the layer zip to each region once, it published to us-east-1 every
time, resulting in:

```
arn:aws:lambda:us-east-1:267093732750:layer:elastic-apm-node-ver-4-7-1:1
arn:aws:lambda:us-east-1:267093732750:layer:elastic-apm-node-ver-4-7-1:2
arn:aws:lambda:us-east-1:267093732750:layer:elastic-apm-node-ver-4-7-1:3
...
arn:aws:lambda:us-east-1:267093732750:layer:elastic-apm-node-ver-4-7-1:22
```

## tl;dr 

This happened because:
1. PR #4109 changed the "release.yml" workflow to use `uses: elastic/oblt-actions/aws/auth@...` for AWS auth, from a different auth mechanism. This new mechanism ultimately uses `aws-actions/configure-aws-credentials` which sets the `AWS_DEFAULT_REGION` and `AWS_REGION` envvars (by default to `us-east-1`).
2. And our publishing step (https://github.com/elastic/apm-agent-nodejs/blob/905289e809bd5675096e596f38d53e1743ad8b9a/.ci/Makefile#L51-L64) sets `AWS_DEFAULT_REGION={intended-region}` and calls `aws ...`, expecting that intended-region to get used.

The problem is that `AWS_REGION` wins here. https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html says:

> AWS_REGION
>
>    The AWS SDK compatible environment variable that specifies the AWS Region to send the request to.
>
>    If defined, this environment variable overrides the values in the environment variable AWS_DEFAULT_REGION and the profile setting region. You can override this environment variable by using the --region command line parameter.


## the fix

The `.ci/Makefile` that is doing the publishing via `aws` CLI commands is using
`AWS_DEFAULT_REGION` to set the region. It can switch to explicitly passing the `--region ...` argument to `aws` commands so that a value of `AWS_REGION` will not get used.

## details

The "Publish AWS lambda" step of the release.yml job for the v4.7.1 release shows `AWS_REGION` and `AWS_DEFAULT_REGION` being set:
https://github.com/elastic/apm-agent-nodejs/actions/runs/10079661414/job/27867507125#step:16:31
The step before (elastic/oblt-actions/aws/auth) set these envvars.

Comparing to the release.yml job for the (correct) v4.7.0 release shows that these AWS envvars were not set:
https://github.com/elastic/apm-agent-nodejs/actions/runs/9503649669/job/26194550532#step:15:31


## aside apm-agent-java

Aside: I'm guessing the same thing happened for apm-agent-java a long while back.
https://github.com/elastic/apm-agent-java/pull/3185 says:

> The lambda layer publish script did not work as expected locally and presumably in GH actions and created 20+ lambda layer versions in eu-west-1 instead of 1 version in every region. 

That PR moved apm-agent-java away from using the equivalent .ci/Makefile code, in favour of https://github.com/elastic/apm-agent-java/blob/af4b613b0a1b63ce2687e00b8cbbf67d2f442810/.ci/publish-aws.sh#L33 that explictly sets `--region ...`.

**Comment by trentm** [comment]
I've replaced the Lambda Layer ARN table on https://github.com/elastic/apm-agent-nodejs/releases/tag/v4.7.1 with a known issue comment.
